MiniSpace.Web/src/MiniSpace.Web/Pages/Comments/CommentsTab.razor

```diff
@@ -0,0 +1,339 @@
+@using Blazorise.DeepCloner
+@using MiniSpace.Web.Areas.Comments
+@using MiniSpace.Web.DTO
+@using MiniSpace.Web.DTO.Wrappers
+@using MiniSpace.Web.Models.Comments
+@using MiniSpace.Web.Pages.Comments.Dialogs
+@using Radzen
+@using AlignItems = Radzen.AlignItems
+@using DialogOptions = Radzen.DialogOptions
+@using DialogService = Radzen.DialogService
+@using Orientation = Radzen.Orientation
+@inject DialogService DialogService
+@inject ICommentsService CommentsService
+
+<RadzenStack Gap="1rem" Orientation="Orientation.Vertical" JustifyContent="JustifyContent.SpaceBetween" Style="height: 100%;">
+    <RadzenStack Orientation="Orientation.Horizontal" JustifyContent="JustifyContent.Center" Gap="0.5rem">
+        <RadzenButton Size="ButtonSize.Medium" Text="Create comment" ButtonStyle="ButtonStyle.Secondary"
+                      Click="@(() => CreateRootComment())" />
+        <RadzenButton Size="ButtonSize.Medium" Text="Sort by criteria"
+                      Click="@OpenSearchCommentsDialog" />
+    </RadzenStack>
+    <RadzenRow AlignItems="AlignItems.Center">
+        <RadzenColumn Size="12" SizeMD="12">
+            @if (pageInitialized && commentsEmpty)
+            {
+                <h3>No comments have been added by students yet.</h3>
+            }
+            <CommentsTree Comments="Comments" StudentId="StudentId"
+                          UpdateComment="@UpdateComment"
+                          DeleteComment="@DeleteComment"
+                          CreateSubComment="@CreateSubComment"
+                          LoadSubComments="@LoadSubComments" 
+                          LoadMoreSubComments="@LoadMoreSubComments"
+                          AddLike="@AddLike"
+                          DeleteLike="@DeleteLike" />
+            @if (moreCommentsAvailable)
+            {
+                <RadzenButton Size="ButtonSize.Medium" Text="Load more comments"
+                              Click="@LoadMoreComments" />
+            }
+        </RadzenColumn>
+    </RadzenRow>
+</RadzenStack>
+
+@code {
+    [Parameter]
+    public string ContextId { get; set; }
+    [Parameter]
+    public string CommentContext { get; set; }
+    [Parameter]
+    public Guid StudentId { get; set; }
+
+    private Guid contextId;
+    private Guid studentId;
+    
+    private HashSet<CommentDto> Comments;
+    
+    private SearchCommentsModel SearchCommentsModel = new()
+    {
+        Pageable = new PageableDto()
+        {
+            Page = 1,
+            Size = 5,
+            Sort = new SortDto()
+            {
+                SortBy = new List<string>() { "lastUpdatedAt" },
+                Direction = "des"
```

Comment:
I think it might be better if parent comments were displayed in des direction but children in asc direction. That way the children comments would be displayed chronologically from top to bottom

Reply:
Also you might want to sort parent comments by "LastReplyAt".

Reply:
After some changes, we have possibilities to choose sort direction for parent comments (des as default) and separetely for child comments (asc as default). Sorting parameters will be common, as default we have "lastReplyAt" and "lastUpdatedAt" (the first one has no impact for child comments so they in fact will be sorted by "lastUpdatedAt").